tfx/components/evaluator/executor.py

```diff
@@ -176,52 +180,60 @@ def Do(self, input_dict: Dict[Text, List[types.Artifact]],
           eval_saved_model_path=model_path,
           add_metrics_callbacks=add_metrics_callbacks))
 
-    file_pattern = io_utils.all_files_pattern(
-        artifact_utils.get_split_uri(input_dict[constants.EXAMPLES_KEY], 'eval')
-    )
     eval_shared_model = models[0] if len(models) == 1 else models
     schema = None
     if constants.SCHEMA_KEY in input_dict:
       schema = io_utils.SchemaReader().read(
           io_utils.get_only_uri_in_dir(
               artifact_utils.get_single_uri(input_dict[constants.SCHEMA_KEY])))
 
+    # Load and deserialize example splits from execution properties.
+    example_splits = json_utils.loads(
+        exec_properties.get(constants.EXAMPLE_SPLITS_KEY, 'null')) or []
+    if not isinstance(example_splits, list):
+      raise ValueError('example_splits in execution properties needs to be a '
+                       'list. Got %s instead.' % type(example_splits))
+
     absl.logging.info('Evaluating model.')
     with self._make_beam_pipeline() as pipeline:
+      examples_list = []
+      tensor_adapter_config = None
       # pylint: disable=expression-not-assigned
-      if _USE_TFXIO:
-        tensor_adapter_config = None
-        if tfma.is_batched_input(eval_shared_model, eval_config):
+      if _USE_TFXIO and tfma.is_batched_input(eval_shared_model, eval_config):
```

Comment:
We seem to be missing the case where _USE_TFXIO is True, but tfma.is_batched_input(eval_shared_model, eval_config) is False.

Reply:
that case is combined in the else branch, because when _USE_TFXIO is True but is_batched_input is False, tensor_adapter_config is still None, which is equivalent to the last else branch. so we decided to merge them.